src/index.ts

```diff
@@ -76,6 +76,7 @@ export interface RegisterOptions {
     logger?: Logger
     transport: () => Promise<LSPConnection> | LSPConnection
     documentSelector: DocumentSelector
+    initializationOptions?: (url: URL) => any
```

Comment:
It feels weird to have a parameter named `initializationOptions` that is a function. Could you rename it `getInitializationOptions()`? Could you also add a docstring explaining why it's needed?

Reply:
Could it return a more specific type than `any`?